flocker/node/agents/ebs.py

```diff
@@ -1245,6 +1254,11 @@ def detach_volume(self, blockdevice_id):
         if ebs_volume.state != VolumeStates.IN_USE.value:
             raise UnattachedVolume(blockdevice_id)
 
+        if ebs_volume.attachments:
+            attachment_state = ebs_volume.attachments[0]['State']
```

Comment:
Why are we only looking at the first attachment? What happens if the first attachment isn't busy but one of the others are?